crates/plugins/mediator-coordination/src/didcomm/bridge.rs

```diff
@@ -36,28 +37,53 @@ impl DIDResolver for LocalDIDResolver {
             return Ok(Some(self.diddoc.clone()));
         }
 
-        if !did.starts_with("did:key:") {
-            return Err(Error::new(
+        if did.starts_with("did:peer:") {
+            // Adding logic to resolve Peer DIDs
+            let method = DidPeer::new_with_format(PublicKeyFormat::Jwk); // Assuming a Peer DID resolver
+            match method.expand(did) {
+                Ok(diddoc) => {
+                    // Attempt to convert DID document representation, propagating errors
+                    match serde_json::from_value(json!(Document {
+                        service: Some(vec![]),
```

Comment:
I recommend implementing `From<{DID method use}> for DIDDoc` directly for a more appropriate conversion of the current `Document` to `DIDDoc`. This eliminates the need to manually re-add the service field as done here, allowing for more precise field population and reducing reliance on `serde_json` for deserialization, which does not correctly parse the document resolved by Peer DID. You can refer to my implementation in the demo branch:
https://github.com/adorsys/didcomm-mediator-rs/blob/2364706ce7fd515ec1b485b34962e9af32d899fc/crates/plugins/mediator-coordination/src/didcomm/bridge.rs#L54-L59
And that's all.

Reply:
Ok let me see to it